#510 Remove plugin exports and update docs

ryanolee opened the pull request:
# What
Removing plugin exports as discussed in #510 

# Why 
So that plugin code is not too tightly coupled to custom plugin implementations

# How
Remove plugin exports and updated documentation 

# Notes
Note: This should not be merged until V6

ryanolee commented:
Will fix the tests relating to this later today / tomorrow (time permitting 😆)

s0ph1e commented on README.md:
```diff
@@ -204,20 +204,43 @@ Number, maximum amount of concurrent requests. Defaults to `Infinity`.
-    * [saveResource](#saveresource)
-    * [generateFilename](#generatefilename)
-    * [getReference](#getreference)
+- [website-scraper](#website-scraper)
```
There are too many changes here, could you please leave only plugins-related items?

s0ph1e commented on README.md:
```diff
@@ -204,20 +204,43 @@ Number, maximum amount of concurrent requests. Defaults to `Infinity`.
+  - [Log and debug](#log-and-debug)
 
 ##### Built-in plugins
 Scraper has built-in plugins which are used by default if not overwritten with custom plugins. You can find them in [lib/plugins](https://github.com/website-scraper/node-website-scraper/tree/master/lib/plugins) directory or get them using 
```
I think the documentation also needs an update here

s0ph1e reviewed (changes requested):
Hi @ryanolee

Thank you for creating the PR. Please fix the failing tests and update the readme according to the comments

ryanolee commented:
Hi @s0ph1e ,
Thanks for the reply! I think the extra changes the readme files came from one of my IDE extensions that annoyingly seems to run on pre commit. Removed the change for now and fixed up the tests / docs. Not sure what we want to do with this PR as it probably does not need to go out until v6 of the scraper. 👀

s0ph1e commented:
Hey @ryanolee,

Thank you for the quick changes. I'm going to merge the PR to master (master stands for current WIP for the next version, v5 code will be in branch 5.x)
